test/e2e/flask/multichain-api/wallet_revokeSession.spec.ts

```diff
@@ -0,0 +1,159 @@
+import { strict as assert } from 'assert';
+import { isObject, pick } from 'lodash';
+import {
+  ACCOUNT_1,
+  ACCOUNT_2,
+  largeDelayMs,
+  WINDOW_TITLES,
+  withFixtures,
+} from '../../helpers';
+import FixtureBuilder from '../../fixture-builder';
+import {
+  initCreateSessionScopes,
+  DEFAULT_MULTICHAIN_TEST_DAPP_FIXTURE_OPTIONS,
+  openMultichainDappAndConnectWalletWithExternallyConnectable,
+  addAccountInWalletAndAuthorize,
+  getSessionScopes,
+  type FixtureCallbackArgs,
+} from './testHelpers';
+
+describe('Initializing a session w/ several scopes and accounts, then calling `wallet_revokeSession`', function () {
+  const GANACHE_SCOPES = ['eip155:1337', 'eip155:1338', 'eip155:1000'];
+  const ACCOUNTS = [ACCOUNT_1, ACCOUNT_2];
+  it('Should return empty object from `wallet_getSession` call', async function () {
+    await withFixtures(
+      {
+        title: this.test?.fullTitle(),
+        fixtures: new FixtureBuilder()
+          .withNetworkControllerTripleGanache()
+          .build(),
+        ...DEFAULT_MULTICHAIN_TEST_DAPP_FIXTURE_OPTIONS,
+      },
+      async ({ driver, extensionId }: FixtureCallbackArgs) => {
+        await openMultichainDappAndConnectWalletWithExternallyConnectable(
+          driver,
+          extensionId,
+        );
+        await initCreateSessionScopes(driver, GANACHE_SCOPES, ACCOUNTS);
+        await addAccountInWalletAndAuthorize(driver);
+        await driver.clickElement({ text: 'Connect', tag: 'button' });
+        await driver.delay(largeDelayMs);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.MultichainTestDApp);
+
+        /**
+         * We verify that scopes are not empty before calling `wallet_revokeSession`
+         */
+        const { sessionScopes } = await getSessionScopes(driver);
+        for (const scope of GANACHE_SCOPES) {
+          assert.ok(
+            isObject(sessionScopes[scope]),
```

Comment:
Do we need to verify that they have keys beyond just being an object?

Reply:
mmm... probably not necessary. Honestly we can probably make this check even less precise by just checking that `Object.keys(sessionScopes).length > 0` instead

Reply:
https://github.com/MetaMask/metamask-extension/pull/27782/commits/22e6e6be21826aa363a36ab98161205f28231893